### PR TITLE
update mongodb readmodel repository implementation

### PR DESCRIPTION
**Pull request by simsim2006**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
         "php": ">=7.0",
         "broadway/broadway": "^2.0",
         "mongodb/mongodb": "^1.0",
-        "beberlei/assert": "^2.7"
+        "beberlei/assert": "^3.0"
```
Comment (simsim2006) [inline composer.json]: it was conflicting with the dependency on broadway core library: https://github.com/broadway/broadway/blob/master/composer.json#L7

File: src/MongoDBRepository.php [hunk 2]
```diff
@@ -39,19 +39,19 @@ public function __construct(Collection $collection, Serializer $serializer, stri
...
      */
-    public function save(Identifiable $model)
+    public function save(Identifiable $model): void
     {
         Assert::isInstanceOf($model, $this->class);
 
         $normalized = $this->normalizeIdentifiable($model);
 
-        $this->collection->insertOne($normalized);
+        $this->collection->replaceOne(['_id' => $model->getId()], $normalized, ['upsert' => true]);
```
Comment (simsim2006) [inline src/MongoDBRepository.php]: I guessed that the idea was to have only 1 method that can be used to save a freshly new ReadModel but also to update an existing ReadModel. That's why it seems to insertOne() should be converted to replaceOne() with upsert flag

**Comment by robinvdvleuten** [comment]
@simsim2006 looks good to me, but you'll have to remove the return types as Travis still testing on PHP 7.0. And removing php 7.0 support is out of scope for this PR.

**Comment by clarkstachio** [comment]
When is this expected to be merged? I'm having the same issue with the beberlei/assert version conflict.

**Pull request closed without merging**